chrome/content/zotero/xpcom/openurl.js

```diff
@@ -203,25 +203,33 @@ Zotero.OpenURL = new function() {
 			}
 		}
 		
-		if(item.creators && item.creators.length) {
-			// encode first author as first and last
-			var firstCreator = item.creators[0];
-			if(item.itemType == "patent") {
-				_mapTag(firstCreator.firstName, "invfirst");
-				_mapTag(firstCreator.lastName, "invlast");
-			} else {
-				if(firstCreator.isInstitution) {
-					_mapTag(firstCreator.lastName, "aucorp");
-				} else {
-					_mapTag(firstCreator.firstName, "aufirst");
-					_mapTag(firstCreator.lastName, "aulast");
+		var first = true;
+		for(var i=0; i<item.creators.length; i++) {
+			var creator = item.creators[i];
+			// only export authors
+			// other creator types cannot correctly export to COinS
+			if(creator.creatorType == "author") {
+				// encode first author as aufirst and aulast
+				if (first) {
+					if(item.itemType == "patent") {
+						_mapTag(creator.firstName, "invfirst");
+						_mapTag(creator.lastName, "invlast");
+					} else {
+						_mapTag(creator.firstName, "aufirst");
+						_mapTag(creator.lastName, "aulast");
+					}
+					first = false;
 				}
-			}
-			
-			// encode subsequent creators as au
-			for(var i=0; i<item.creators.length; i++) {
-				_mapTag((item.creators[i].firstName ? item.creators[i].firstName+" " : "")+
-					item.creators[i].lastName, (item.itemType == "patent" ? "inventor" : "au"));
+				// encode all authors (first and subsequents) as 
+				// au, aucorp or inventor
+				var goal = "au";
+				if (creator.fieldMode == 1) {
+					goal = "aucorp";
+				}
+				if (item.itemType == "patent") {
+					goal = "inventor";
+				}
+				_mapTag((creator.firstName ? creator.firstName+" " : "")+creator.lastName, goal);
```

Comment:
also, while we have been exporting in "first last" format, I think "last, first" is much better. Both formats are given as examples in the [OpenURL documentation](http://www.niso.org/kst/reports/standards/kfile_download?id%3Austring%3Aiso-8859-1=Z39-88-2004.pdf&pt=RkGKiXzW643YeUaYUqZ1BFwDhIG4-24RJbcZBWg8uE4vWdpZsJDs4RjLz0t90_d5_ymGsj_IKVaGZww13HuDlZQ8NBt1sTxP_v4iiGqH7rSaAeVDnMfeKJrrJ_JSEGPB) (page 37)